server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -710,6 +753,36 @@ public static void validateRepositoryMetadataSettings(
         }
     }
 
+    private static boolean repositoryWithShallowV2Exists(Map<String, Repository> repositories) {
+        return repositories.values().stream().anyMatch(repo -> SHALLOW_SNAPSHOT_V2.get(repo.getMetadata().settings()));
+    }
+
+    private static boolean pinnedTimestampExistsWithDifferentRepository(
+        String newRepoName,
+        Settings settings,
+        RepositoriesService repositoriesService
+    ) throws IOException {
+        Map<String, Set<Long>> pinningEntityTimestampMap = RemoteStorePinnedTimestampService.fetchPinnedTimestamps(
+            settings,
+            repositoriesService
+        );
+        for (String pinningEntiy : pinningEntityTimestampMap.keySet()) {
+            String[] tokens = pinningEntiy.split(SnapshotsService.SNAPSHOT_PINNED_TIMESTAMP_DELIMITER);
+            if (tokens.length > 2) {
```

Comment:
This makes things a bit complicated. I am thinking of adding a restriction on repo name for shallow snapshot V2 to not contain `SNAPSHOT_PINNED_TIMESTAMP_DELIMITER `

Reply:
Made changes in the second commit.